cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,553 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+DOCUMENTATION = '''
+module: cloudfront
+short_description: Create, delete and update actions for the main AWS
+  Cloudfront actions.
+description:
+  - Read the AWS documentation for Cloudfront for the correct json values
+      as there are too many to list here.
+  - Creates distributions, streaming_distributions, invalidations,
+      origin_access_ids
+  - Deletes distributions, streaming_distributions, origin_access_ids
+  - Updates distributions, streaming_distributions
+version_added: "2.1"
+requirements: [ boto3 ]
+options:
+  type:
+    description:
+      - specifies the resource to take action upon, streaming is
+        streaming_distribution
+    required: True
+    choices: [
+            'distribution',
+            'origin_access_id',
+            'invalidation',
+            'streaming',
+            ]
+    default: None
+  policy:
```

Comment:
For usability purposes I think this would be better broken out in to policy_file and policy_string, set as mutually exclusive parameters.

I spent the best part of a day getting my task working as I had a one character error in my json template which meant it was trying to get loaded as a file.

Obviously entirely my own fault but if separate parameters were used then we could return a much more explicit "Invalid JSON" error message, which would have saved me a few hours :)